src/reportSources/legacyFflogs/eventAdapter/__tests__/deduplicateAoE.ts

```diff
@@ -0,0 +1,258 @@
+
+const actor: Actor = {
+	id: '1',
+	kind: '1',
+	name: 'Test Actor',
+	team: Team.FRIEND,
+	playerControlled: true,
+	job: 'UNKNOWN',
+}
+
+const timestamp = 1600000000000
+
+const pull: Pull = {
+	id: '1',
+	timestamp,
+	duration: 1,
+	encounter: {
+		name: 'test encounter',
+		duty: {
+			id: 1,
+			name: 'test duty',
+		},
+	},
+	actors: [actor],
+}
+
+const report: Report = {
+	timestamp,
+	edition: GameEdition.GLOBAL,
+	name: 'Event adapter test',
+	pulls: [pull],
+	meta: {
+		source: 'legacyFflogs',
+		code: 'adapterTest',
+		loading: false,
+		enemies: [],
+		enemyPets: [],
+		fights: [],
+		friendlies: [],
+		friendlyPets: [],
+		lang: ReportLanguage.ENGLISH,
+		phases: [],
+		end: Infinity,
+		owner: 'test',
+		start: 0,
+		title: 'Event adapter test',
+		zone: 0,
+	},
+}
+
+const eventAttrs : { cause: Cause, source: string }= {
+	cause: {
+		type: 'action',
+		action: 1,
+	},
+	source: 's1',
+}
+
+const damageTargetAttrs = {
+	amount: 1,
+	overkill: 0,
+	sourceModifier: SourceModifier.NORMAL,
+	targetModifier: TargetModifier.NORMAL,
+	bonusPercent: 0,
+}
+
+const healTargetAttrs = {
+	amount: 1,
+	overheal: 0,
+	sourceModifier: SourceModifier.NORMAL,
+}
+
+function ensureOrder(timestamps: number[]): void {
+	for (let i = 0; i < timestamps.length; i++) {
+		if (i === 0) {
+			continue
+		}
+		if (timestamps[i] < timestamps[i - 1]) {
+			fail()
+		}
+	}
+}
+
+describe('DeduplicateAoEStep', () => {
+	let step: DeduplicateAoEStep
+	beforeEach(() => {
+		step = new DeduplicateAoEStep({
+			report,
+			pull,
+		})
+	})
+	describe('#postprocess', () => {
+		describe('when dealing with damage events', () => {
+			const events: Event[] = [
+				{
+					type: 'damage',
+					...eventAttrs,
+					timestamp: 1,
+					sequence: 1,
+					targets: [{
+						target: 't1',
+						...damageTargetAttrs,
+					}],
+				},
+				{
+					type: 'damage',
+					...eventAttrs,
+					timestamp: 1,
+					sequence: 1,
+					targets: [{
+						target: 't2',
+						...damageTargetAttrs,
+					}],
+				},
+				{
+					type: 'damage',
+					...eventAttrs,
+					timestamp: 2,
+					targets: [{
+						target: 't1',
+						...damageTargetAttrs,
+					}],
+				},
+				{
+					type: 'damage',
+					...eventAttrs,
+					timestamp: 2,
+					targets: [{
+						target: 't2',
+						...damageTargetAttrs,
+					}],
+				},
+			]
+
+			it('deduplicates damage events', () => {
+				const results = step.postprocess(events)
+
+				// eslint-disable-next-line @typescript-eslint/no-magic-numbers
+				expect(results.length).toEqual(3)
+				expect((results[0] as Events['damage']).targets.length).toEqual(2)
+			})
+			it('does not deduplicate DoT events', () => {
+				const results = step.postprocess(events)
+
+				expect(results.filter(e => (e as Events['damage']).sequence != null).length).toEqual(1)
+			})
+			it('retains order of events', () => {
+				const results = step.postprocess(events)
+
+				ensureOrder(results.map(e => e.timestamp))
```

Comment:
This needs some additional logic to ensure that the two DOT events didn't get scrambled by the ordering, since their timestamps are both the same.